microk8s-resources/default-args/etcd

```diff
@@ -5,3 +5,4 @@
 --trusted-ca-file=${SNAP_DATA}/certs/ca.crt
 --cert-file=${SNAP_DATA}/certs/server.crt
 --key-file=${SNAP_DATA}/certs/server.key
+--enable-v2=true
```

Comment:
I'm curious why an upgrade will exclude the contents of this file. Wouldn't the upgrade restart etcd and read the default args file from the contents of the new snap?

Reply:
During installation in the install hook (https://github.com/ubuntu/microk8s/blob/master/snap/hooks/install#L12) we copy the default arguments to a location where the user has write access. As the user may tune the configuration arguments and/or he is used to working with a specific k8s setup we do not touch these arguments during upgrades,unless we really have to. This is a "we really have to" case because by upgrading etcd a new argument is needed.

Reply:
used util to add option to configure hook